x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java

```diff
@@ -93,25 +96,24 @@ public void process(NativePyTorchProcess process) {
         logger.debug(() -> new ParameterizedMessage("[{}] Results processing finished", deploymentId));
     }
 
-    public synchronized LongSummaryStatistics getTimingStats() {
-        return new LongSummaryStatistics(timingStats.getCount(), timingStats.getMin(), timingStats.getMax(), timingStats.getSum());
+    public synchronized ResultStats getResultStats() {
+        return new ResultStats(
+            new LongSummaryStatistics(timingStats.getCount(), timingStats.getMin(), timingStats.getMax(), timingStats.getSum()),
+            errorCount,
+            pendingResults.size(),
+            lastUsed
+        );
     }
 
     private synchronized void processResult(PyTorchResult result) {
         if (result.isError() == false) {
             timingStats.accept(result.getTimeMs());
             lastUsed = Instant.now();
+        } else {
+            errorCount++;
```

Comment:
Do we want to count the following as errors as we well?

 - Queue rejection
 - Timeout
 - missing field?

Reply:
Yes!

Reply:
OK, then the error count cannot only be in the process result manager :(. Might complicate things.